infrastructure/src/main/java/org/corfudb/infrastructure/CompactorService.java

```diff
@@ -135,51 +189,51 @@ private void runOrchestrator() {
             boolean isLeader = isNodePrimarySequencer(updateLayoutAndGet());
             log.trace("Current node isLeader: {}", isLeader);
 
+            CompactorLeaderServices compactorLeaderServices = getCompactorLeaderServices();
```

Comment:
I don't see the `shutdown()` method of `CompactorService` resetting the `optionalCompactorLeaderServices` which means this call can potentially re-use the old instance right in which case this issue can still reoccur right? Please correct me if I am missing something?

Reply:
L264 does reset the optionalCompactorLeaderServices. So This line shouldn't be returning the old instance